src/window.rs

```diff
@@ -286,13 +311,20 @@ impl WindowUtil for Window {
     }
 }
 
-fn calculate_max_repair(num_peers: u64, consumed: u64, received: u64, times: usize) -> u64 {
+fn calculate_max_repair(
+    num_peers: u64,
+    consumed: u64,
+    received: u64,
+    times: usize,
+    is_next_leader: bool,
+) -> u64 {
     // Calculate the highest blob index that this node should have already received
     // via avalanche. The avalanche splits data stream into nodes and each node retransmits
     // the data to their peer nodes. So there's a possibility that a blob (with index lower
     // than current received index) is being retransmitted by a peer node.
-    let max_repair = if times >= 8 {
-        // if repair backoff is getting high, don't wait for avalanche
+    let max_repair = if times >= 8 || is_next_leader {
```

Comment:
Instead of changing this function, would it make sense to call it with zero `num_peers` when `is_next_leader`? I can see it'd give you the same result, which is appealing, but is that a hack or a meaningful use of `num_peers` when a node is a leader?  Maybe check with @pgarg66?